src/IECore/MeshAlgo.cpp

```diff
@@ -225,5 +447,117 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents(
 	return std::make_pair( tangentPrimVar, bitangentPrimVar );
 }
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation )
+{
+	PrimitiveVariableMap::const_iterator it = mesh->variables.find( primitiveVariable );
+
+	if( it == mesh->variables.end() )
+	{
+		return PrimitiveVariable();
+	}
+
+	Data *srcData = it->second.data.get();
+	DataPtr dstData;
+
+	PrimitiveVariable::Interpolation srcInterpolation = it->second.interpolation;
+
+	if ( srcInterpolation == interpolation )
+	{
+		return it->second;
+	}
+
+	// average array to single value
+	if ( interpolation == PrimitiveVariable::Constant )
+	{
+		AverageValueFromVector fn;
+		dstData = despatchTypedData<AverageValueFromVector, IsArithmeticVectorTypedData>( srcData, fn );
+		return PrimitiveVariable( interpolation, dstData );
+	}
+
+	// splat single value into array
+	if ( srcInterpolation == PrimitiveVariable::Constant )
+	{
+		size_t len = mesh->variableSize( interpolation );
+		switch( srcData->typeId() )
```

Comment:
No reason. Just the way it was before. I'll add a TODO to update this. The spat scalar to array function will be required in CurveAlgo & PointsAlgo versions of resamplePrimitiveVariable also so I could do it then. Or would you prefer it's done now?